terminal.go

```diff
@@ -211,15 +211,22 @@ func (g *LightningTerminal) Run() error {
 		return fmt.Errorf("could not create permissions manager")
 	}
 
+	// Construct the rpcProxy. It must be initialised before the main web
+	// server is started.
+	g.rpcProxy = newRpcProxy(g.cfg, g, g.validateSuperMacaroon, g.permsMgr)
+
+	// Start the main web server that dispatches requests either to the
+	// static UI file server or the RPC proxy. This makes it possible to
+	// unlock lnd through the UI.
+	if err := g.startMainWebServer(); err != nil {
```

Comment:
Since we start the main web server this early, we have to guard the `g.restHandler` from being called if it's nil (. Otherwise a REST request coming in too early would panic. Or we move the call to `createRESTProxy()` to before this call even.

Reply:
Hmm, no, creating the REST proxy before the main web server is started will probably not work since the REST proxy will attempt to connect to the main web server on creation.

Reply:
That makes me wonder if we could extract the main web server into its own struct that owns the REST proxy and RPC proxy (and probably also the status server)? Then the start/stop would be a bit more abstracted as well. But not sure how large of a diff that is.